security/fma-withdrawals-root-in-block-header.md

```diff
@@ -0,0 +1,133 @@
+
+### FM1: Withdrawals downtime do to inaccurate `withdrawalsRoot`
+
+- **Description:**
+  If the `withdrawalsRoot` in the block header is incorrect, critical infra used to enable withdrawals may fail. Namely, output proposals and challenges would be incorrect, affecting chains with permissioned and chains with permissionless proofs. This is because these components will, with the activation of the Isthmus fork, use the `withdrawalsRoot` header field instead of querying the information from an archive node in the usual way. Output roots are returned by the op-node [`optimism_outputAtBlock`](https://docs.optimism.io/operators/node-operators/json-rpc#optimism_outputatblock) RPC method, and this behaves differently under Isthmus -- when handling a request for an output root (it no longer delegates an `eth_getProof` to op-geth and just reads the information from the block header).
+
+  Triggers:
+
+  - A failed hardfork activation in the execution client.
+
+  - If there is an execution client bug, for example it is possible the root is (incorrectly) added to the header before the state is fully committed.
+
+  - If we were to ever introduce non empty `withdrawals` in the block body, this might override the mechanism introduced with this feature and invalidate the interpreation of the `withdrawalsRoot` field.
+
+- **Risk Assessment:**
+
+  **High impact, low likelihood**
+
+  Temporary downtime for withdrawals, or loss of funds if not remediated in time to challenge a malicious proposal.
+
+  **Mitigations:**
+
+  - We rely on e2e tests to check for consistency between the outputs returned from op-node and those constructed manually in the old way.
+
+  - Instead of waiting for the failure mode to materialize and then writing a patch in a rush, we could add an optional config var to op-node to switch it back into the old behavior. Rolling out the fix would not then require any software releases.
+
+  - op-geth could be modified to log a critical error (triggering an alert) if the withdrawals list in the body is ever non empty.
+
+- **Detection:**
+  Fault proof monitoring systems may not detect this failure mode immediately, until an actor running patched software made a proposal or challenge.
+
+- **Recovery Path(s)**:
+  Fault proof infra would nee to be pointed at a patched op-node. The patch would restore the old behaviour for generating output roots.
+
+### FM2: Chain split from genesis due to divergent genesis block computation
+
+- **Description:**
+  The genesis state is, with the activation of Isthmus at genesis, now part of the genesis block. Therefore a consensus bug where clients compute a different genesis block from the same genesis state can cause a chain split.
+
+- **Risk Assessment:**
+
+  **High impact, low likelihood**
+
+  **Mitigations:**
+  This should be caught by cross-client testing (kurtosis and devnets)
+
+- **Detection:**
+  If the bug made it to production, replica healthcheck alerts would fire as nodes diverged.
```

Comment:
This is clearer, but I would like to know at what moment we will receive this alerts in the process. 
Can we just add at what moment the node would be "tested" and then will fire alerts?

Reply:
we do have alerting in prod for observing chain splits. the chances of a bug that come from a chainsplit after observing a few correct blocks is very small, because it would imply something about the state root calculation is incorrect, which would result in a chain split anyways. so i think just seeing that a few blocks with different withdrawal roots across multiple clients should give confidence that this will not result in a chain split

Reply:
As discussed on discord, we can use a combination of:
1. Routing alerts from devnets and doing manual inspections of metrics and
2. Adding an acceptance test which validates no chain split after a few blocks

Reply:
Have added action items for these, @Ethnical please resolve the thread if you are happy.

Reply:
Thanks!